src/Microsoft.SqlTools.ServiceLayer/QueryExecution/DataStorage/ServiceBufferFileStreamReader.cs

```diff
@@ -134,14 +134,15 @@ public IList<DbCellValue> ReadRow(long fileOffset, long rowId, IEnumerable<DbCol
                         continue;
                     }
 
-                    // The typename is stored in the string
+                    // We need to specify the assembly name for SQL types in order to resolve the type correctly.
+                    if (sqlVariantType.StartsWith("System.Data.SqlTypes."))
+                    {
+                        sqlVariantType = sqlVariantType + ", System.Data.Common";
```

Comment:
Is this code from SSMS or something? how are they handling it otherwise?

Reply:
similar to SSMS with modification, but this specific line, SSMS doesn't have to do this.

Reply:
as per the microsoft doc:
![image](https://user-images.githubusercontent.com/13777222/145139826-49732c33-5f54-410a-918e-64bdabf3dbd6.png)

it should be full qualified name, not sure why SSMS doesn't need it, I can try to see whether .net 4 and .net 5 makes a difference.

Reply:
.net 4.7 and .net 5 all behave the same, null is returned is only the first part is provided, I don't have a SSMS debug environment, I can ask Matteo to debug and share the findings.